samples/createSubscriptionWithCloudStorage.js

```diff
@@ -0,0 +1,99 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This is a generated sample, using the typeless sample bot. Please
+// look for the source TypeScript sample (.ts) for modifications.
+'use strict';
+
+/**
+ * This snippet demonstrates how to perform basic operations on
+ * schemas with the Google Cloud Pub/Sub API.
+ *
+ * For more information, see the README.md under /pubsub and the documentation
+ * at https://cloud.google.com/pubsub/docs.
+ */
+
+// sample-metadata:
+//   title: Create a Cloud Storage subscription
+//   description: Demonstrates how to create a subscription with Cloud Storage.
+//   usage: node createSubscriptionWithCloudStorage.js <topic-name> <subscription-name> <bucket> <filenamePrefix> <filenameSuffix> <maxDuration>
+
+// [START pubsub_create_cloud_storage_subscription]
+/**
+ * TODO(developer): Uncomment these variables before running the sample.
+ */
+// const topicName = 'YOUR_TOPIC_NAME';
+// const subscriptionName = 'YOUR_SUBSCRIPTION_NAME';
+// const bucket = 'YOUR_BUCKET_ID';
+// const filenamePrefix = 'YOUR_FILENAME_PREFIX';
+// const filenameSuffix = 'YOUR_FILENAME_SUFFIX';
+// const maxDuration = 60;
+
+// Imports the Google Cloud client library
+const {PubSub} = require('@google-cloud/pubsub');
+
+// Creates a client; cache this for further use
+const pubSubClient = new PubSub();
+
+async function createSubscriptionWithCloudStorage(
+  topicName,
+  subscriptionName,
+  bucket,
+  filenamePrefix,
+  filenameSuffix,
+  maxDuration
+) {
+  const options = {
+    cloudStorageConfig: {
+      bucket,
+      filenamePrefix,
+      filenameSuffix,
+      maxDuration: {
+        seconds: maxDuration,
+      },
+    },
+  };
+
+  const [sub] = await pubSubClient
```

Comment:
Optional: I noticed the BQ subscription sample doesn't capture the created sub in a variable: https://cloud.google.com/pubsub/docs/samples/pubsub-create-bigquery-subscription?hl=en#pubsub_create_bigquery_subscription-nodejs_javascript Should we consider making these consistent too? Same suggestion applies to the typescript sample.

Reply:
I don't feel strongly about it, but it looks like the other createSubscription samples in here don't capture it either, so I'll change it.